### PR TITLE
feat: Version data class

### PR DESCRIPTION
**Pull request by Adolar0042**

### CODE REVIEW

**Comment by Adolar0042** [comment]
waiting for @TobyBridle to fix a crash so I can actually test this

**Comment by TobyBridle** [comment]
Sorry for taking so long to fix, but should all be good now :)

**Review by TobyBridle — COMMENTED** [review]
test

File: app/src/main/java/com/chouten/app/domain/use_case/module_use_cases/AddModuleUseCase.kt [hunk 1]
```diff
@@ -338,58 +345,28 @@ class AddModuleUseCase @Inject constructor(
...
-                                    IOException("Could not delete module ${it.second.name} (${it.second.id})"),
-                                    newModuleUri
-                                )
-                            }
-                            log("Updated module ${module.name} (${module.id})")
-                        }
-
-                        // Module being installed is the same version as the existing module
-                        0 -> {
+                    if (module.version < it.second.version) { // old module version < new module version
```
Comment (TobyBridle) [inline app/src/main/java/com/chouten/app/domain/use_case/module_use_cases/AddModuleUseCase.kt]: Here the "old" module is actually newer. We should remove the old version instead.
```suggestion
                    if (module.version < it.second.version) { // old module version > new module version
                         // Delete the old module
                            if (DocumentFile.fromSingleUri(mContext, newModuleUri)?.delete() == false) {
```

File: app/src/main/java/com/chouten/app/domain/use_case/module_use_cases/AddModuleUseCase.kt [hunk 2]
```diff
@@ -338,58 +345,28 @@ class AddModuleUseCase @Inject constructor(
...
+                        // Delete the old module
+                        if (DocumentFile.fromSingleUri(mContext, it.first)?.delete() == false) {
                             safeException(
-                                IllegalArgumentException("Module ${module.name} (${module.id}) already exists"),
-                                newModuleUri
-                            )
-                        }
-
-                        // Module being installed is older than the existing module
```
Comment (TobyBridle) [inline app/src/main/java/com/chouten/app/domain/use_case/module_use_cases/AddModuleUseCase.kt:386]: Module being installed is NEWER than the existing module.
```suggestion
    // Module being installed is newer than the existing module
    else -> {
        safeException(
            IllegalArgumentException("Module ${module.name} (${module.id}) is older than the existing module"),
            IOException("Could not delete module ${it.second.name} (${it.second.id})"),
            it.first
        )
    }
```

**Review by TobyBridle — CHANGES_REQUESTED** [review]

**Comment by TobyBridle** [comment]
Please could you also reformat after submitting the changes

**Comment by TobyBridle** [comment]
LGTM thanks :+1:

**Pull request merged**